x/concentrated-liquidity/position_test.go

```diff
@@ -129,6 +129,22 @@ func (s *KeeperTestSuite) TestInitOrUpdatePosition() {
 
 			// Check that the initialized or updated position matches our expectation
 			s.Require().Equal(test.expectedLiquidity, positionInfo.Liquidity)
+
+			// Check that the relevant uptime accumulators were properly checkpointed
+			positionName := string(types.KeyFullPosition(validPoolId, s.TestAccs[0], test.param.lowerTick, test.param.upperTick, test.param.frozenUntil))
+			uptimeAccums, err := s.App.ConcentratedLiquidityKeeper.GetUptimeAccumulators(s.Ctx, test.param.poolId)
+			s.Require().NoError(err)
+
+			// If frozen for more than a specific uptime's period, the record should exist
+			for uptimeIndex, uptime := range types.SupportedUptimes {
+				recordExists, err := uptimeAccums[uptimeIndex].HasPosition(positionName)
+				s.Require().NoError(err)
+				if test.param.frozenUntil.Sub(s.Ctx.BlockTime()) >= uptime {
+					s.Require().True(recordExists)
+				} else {
+					s.Require().False(recordExists)
+				}
+			}
```

Comment:
nit: It might even be better to hardcoded in a slice for which uptime we expect this to be true. Otherwise, if there is an error in the code, we can have it here left unnoticed

Reply:
I actually thought of this but I intentionally want to keep uptime-related tests agnostic to the number of uptimes we have so that if we change anything it doesn't break a bunch of things. Noted the concern though and will keep this in mind